generators/Exercises/TwelveDays.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Generators.Input;
+using Generators.Output;
+using System.Linq;
+using Humanizer;
+
+namespace Generators.Exercises
+{
+    public class TwelveDays : GeneratorExercise
+    {
+        protected override void UpdateCanonicalData(CanonicalData canonicalData)
+        {
+            canonicalData.Exercise = "twelve-days-song";
+            foreach (var canonicalDataCase in canonicalData.Cases)
+            {
+                canonicalDataCase.UseVariableForExpected = true;
+                var properties = canonicalDataCase.Properties as Dictionary<string, object>;
+                var input = properties["input"] as Dictionary<string, object>;
+                if ((long)input["startVerse"] == (long)input["endVerse"])
+                {
+                    canonicalDataCase.Property = "verse";
+                    properties["input"] = input["startVerse"];
+                }
+                else canonicalDataCase.Property = "verses";
+            }
+        }
+
+        protected override string RenderTestMethodBodyArrange(TestMethodBody testMethodBody)
```

Comment:
This is not the correct method to override, as this is really a core method and you should rarely have to change its behavior. What you are looking for (formatting the multiline string) should be easy to do, see [this generator](https://github.com/exercism/csharp/blob/master/generators/Exercises/FoodChain.cs#L11).

Reply:
@ErikSchierboom I will make the changes as you requested - my previous comments were just to clarify my first version code's background.

Reply:
Well, it *is* still a valid extension point, albeit one that should seldom be used. Thanks for the explanation.

Reply:
@ErikSchierboom OK, [this generator](https://github.com/exercism/csharp/blob/master/generators/Exercises/FoodChain.cs#L11]) doesn't add trailing new line - \n - to the end of verses' strings. Should I then remove the trailing new line - \n - from the Example.cs cases?:

            case 1:
              return "On the first day of Christmas my true love gave to me, a Partridge in a Pear Tree.\n";
          ...

If I will remove the trailing new line - \n - from the end of verses' strings then [twelve-days problems description](https://github.com/exercism/problem-specifications/blob/master/exercises/twelve-days/description.md) should be also corrected AFAIU.

OK. I will make all these related corrections.

Reply:
That is actually a good point! I think you may have found an issue with the problem description, which implies that there should be an empty lines between each line. The canonical data however states otherwise, and in this case I would like to follow the canonical data (so no blank lines between lines, and no trailing newline for a single line.